docs-src/go/chapter-durable-execution/non-deterministic-code-changes.md

```diff
@@ -0,0 +1,70 @@
+---
+id: non-deterministic-code-changes
+title: Non-deterministic code changes
+description: History Replay, sometimes also called Workflow Replay, is the mechanism that Temporal uses to reconstruct the state of a Workflow Execution. Temporal provides Durable Execution via this Replay Functionality.
+sidebar_label: Durability through Replays
+tags:
+  - go sdk
+  - developer-guide-doc-type
+  - event history
+  - replay
+  - durable execution
+---
+
+The most important thing to take away from the section is to make sure you have an application versioning plan whenever you are developing and maintaining a Temporal Application that will eventually deploy to a production environment.
```

Comment:
General comment: This section shows up in the [Develop for Durable Executions section](https://temporal-documentation-git-durable-execution-go.preview.thundergun.io/dev-guide/go/durable-execution#why-replay-test) of the Dev Guide under the "Non-deterministic code changes" heading, which follows a list with two items. It's not immediately clear to which of those two items ("code changes" or "intrinsic non-determinstic logic") this heading pertains. In other words, the heading mixes words from both list items. "How code changes can affect determinism" might be a better heading.

Also, I would recommend covering the "intrinsic non-deterministic logic" case first. It's intuitive that, for example, using a random number in a conditional statement can lead to a different execution path with each invocation. Once you understand why that's a problem, it becomes easier to understand why a code change involving only deterministic statements could break compatibility with Workflow Executions that were execution at the time of deployment.

Reply:
Clarified the list to match headings, and did the ordering switch.